src/Entity/Adherent.php

```diff
@@ -955,12 +934,10 @@ public function setComEmail(?bool $comEmail): void
             $comEmail ? self::CITIZEN_PROJECT_EMAIL_DEFAULT_DISTANCE : self::DISABLED_CITIZEN_PROJECT_EMAIL
         );
 
+        $this->localHostEmailsSubscription = $comEmail;
+
         if ($comEmail) {
-            $subscriptions = [
-                AdherentEmailSubscription::SUBSCRIBED_EMAILS_MAIN,
-                AdherentEmailSubscription::SUBSCRIBED_EMAILS_REFERENTS,
-                AdherentEmailSubscription::SUBSCRIBED_EMAILS_LOCAL_HOST,
-            ];
+            $subscriptions = \call_user_func_array('array_merge_recursive', AdherentEmailSubscription::SUBSCRIPTIONS);
```

Comment:
Please add a static method in `AdherentEmailSubscription` class to encapsulate this behavior. It would avoid copy/paste and would bring some meaningfulness to this piece of code that I barely understand.

Please consider replacing `call_user_func_array` by `...` operator if it's possible.